pytorch_tabnet/tab_network.py

```diff
@@ -405,3 +376,71 @@ def forward(self, x):
         x = self.bn(x)
         out = torch.mul(x[:, :self.output_dim], torch.sigmoid(x[:, self.output_dim:]))
         return out
+
+
+class EmbeddingGenerator(torch.nn.Module):
+    """
+        Classical embeddings generator
+    """
+
+    def __init__(self, input_dim, cat_dims, cat_idxs, cat_emb_dim):
+        """ This is an embedding module for an entier set of features
+
+        Parameters
+        ----------
+        input_dim : int
+            Number of features coming as input (number of columns)
+        cat_dims : list of int
+            Number of modalities for each categorial features
+            If the list is empty, no embeddings will be done
+        cat_idxs : list of int
+            Positional index for each categorical features in inputs
+        cat_emb_dim : int or list of int
+            Embedding dimension for each categorical features
+            If int, the same embdeding dimension will be used for all categorical features
+        """
+        super(EmbeddingGenerator, self).__init__()
+        if cat_dims == []:
+            self.skip_embedding = True
+            self.post_embed_dim = input_dim
+        else:
```

Comment:
I think it would be better to return here in if, instead of making a else (less readable)

Reply:
changed that a bit